src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas2/StatusUpdateService.kt

```diff
@@ -65,13 +81,27 @@ class StatusUpdateService(
       ),
     )
 
+    statusDetails?.forEach { detail ->
+      statusUpdateDetailRepository.save(
+        Cas2StatusUpdateDetailEntity(
+          id = UUID.randomUUID(),
+          statusDetailId = detail.id,
+          statusUpdate = createdStatusUpdate,
+          label = detail.label,
+        ),
+      )
+    }
+
     createStatusUpdatedDomainEvent(createdStatusUpdate)
 
     return AuthorisableActionResult.Success(
       ValidatableActionResult.Success(createdStatusUpdate),
     )
   }
 
+  private fun findStatusDetailOnStatus(status: Cas2PersistedApplicationStatus, detailName: String) =
```

Comment:
could this be a function on the Cas2PersistedApplicationStatus class?

Reply:
nice one, done 👍